### ISSUE TITLE
Use canonical filename for debounce library

### ISSUE THREAD

**Issue by nedbat** [coveragepy/coveragepy#505]
*Originally reported by* **Ben Finney (Bitbucket: [bignose](https://bitbucket.org/bignose), GitHub: [bignose](https://github.com/bignose))**

----------------------------------------

The library “[jQuery throttle / debounce](http://benalman.com/projects/jquery-throttle-debounce-plugin/)” is provided using the canonical filename `jquery.ba-throttle-debounce.js` (or, in compiled form, with a `.min.js` suffix instead).

Please use the canonical name when searching for the library file, to allow easier upgrade and management of the library.

----------------------------------------
- Bitbucket: https://bitbucket.org/ned/coveragepy/issue/505
- This issue had attachments: debounce-library-name.r3008.bundle.  See the [original issue](https://bitbucket.org/ned/coveragepy/issue/505) for details.

**Comment by nedbat** [comment]
*Original comment by* **Ben Finney (Bitbucket: [bignose](https://bitbucket.org/bignose), GitHub: [bignose](https://github.com/bignose))**

----------------------------------------

My suggested solution for this is in the fork at https://vcs.whitetree.org/hg/python/coveragepy/devel/ — revision 3006–3008.

* Name the debounce library to the canonical upstream filename.
* Correct references to the debounce library file.

**Comment by nedbat** [comment]
That vcs.whitetree.org URL seems to have nothing interesting, just a link to a parent directory?

**Comment by nedbat** [comment]
*Original comment by* **Ben Finney (Bitbucket: [benf_wspdigital](https://bitbucket.org/benf_wspdigital), GitHub: Unknown)**

----------------------------------------

> That vcs.whitetree.org URL seems to have nothing interesting

To a web browser, right. But to Mercurial it contains a repository. It will work fine with `hg incoming`, etc.

**Comment by nedbat** [comment]
Tried to clone it:
```
$ hg clone https://vcs.whitetree.org/hg/python/coveragepy/devel/
destination directory: devel
requesting all changes
adding changesets
adding manifests
adding file changes
transaction abort!
rollback completed
abort: missing support for negative part header size: -1
```
Can you put your changes somewhere more accessible?  A pull request perhaps?

**Comment by nedbat** [comment]
*Original comment by* **Ben Finney (Bitbucket: [bignose](https://bitbucket.org/bignose), GitHub: [bignose](https://github.com/bignose))**

----------------------------------------

Here is a Mercurial bundle of the changeset I created to resolve this issue.

You apply a Mercurial bundle using [the `unbundle` command](https://www.mercurial-scm.org/wiki/CommunicatingChanges).

**Comment by nedbat** [comment]
Merged in [7c3dddfe8389 (bb)](https://bitbucket.org/ned/coveragepy/commits/7c3dddfe8389)

**Comment by nedbat** [comment]
Closes issue #505

→ <<cset [7c3dddfe8389 (bb)](https://bitbucket.org/ned/coveragepy/commits/7c3dddfe8389)>>

**Comment by nedbat** [comment]
This fix was released as part of Coverage.py 4.3.
